ui/src/buckets/actions/thunks.ts

```diff
@@ -116,12 +139,10 @@ export const createBucket = (bucket: OwnBucket) => async (
 }
 
 export const updateBucket = (bucket: OwnBucket) => async (
-  dispatch: Dispatch<Action>,
-  getState: GetState
+  dispatch: Dispatch<Action>
 ) => {
   try {
-    const state = getState()
-    const data = denormalizeBucket(state, bucket)
+    const data = denormalizeBucket(bucket)
```

Comment:
Since the purpose of `denormalizeBucket` was to add a labels to the bucket and update the type of bucket to be a different type, this no longer seems relevant

Reply:
I think the Bucket does need to go from `OwnBucket` type (with a labels property) to `GenBucket` type (no labels). If I remove this part then it crashes.

Reply:
totally:

https://github.com/influxdata/influxdb/pull/17688#discussion_r406321100

If that's the case then we should strip the properties that make it an `OwnBucket` so that the type change accurately reflects the type it's being changed to. More specifically, we can ensure that the `labels` and `readableRetention` are removed before forcing the type change